astropy/table/column.py

```diff
@@ -1091,28 +1115,40 @@ def insert(self, obj, values, mask=None):
 
         out = new_ma.view(self.__class__)
         out.parent_table = None
+        out.indices = []
         out._copy_attrs(self)
 
         return out
 
-    def __getitem__(self, item):
-        out = super(MaskedColumn, self).__getitem__(item)
-
+    def copy_attribs(self, out):
         # Fixes issue #3023: when calling getitem with a MaskedArray subclass
         # the original object attributes are not copied.
         if out.__class__ is self.__class__:
             out.parent_table = None
+            # we need this because __getitem__ does a shallow copy of indices
+            if out.indices is self.indices:
+                out.indices = []
             out._copy_attrs(self)
-
         return out
 
+    def __getitem__(self, item):
+        out = super(MaskedColumn, self).__getitem__(item)
+        return self.copy_attribs(out)
+
+    def get_item(self, item):
```

Comment:
This needs a docstring -- and probably a comment about why both it and `__getitem__` exist.